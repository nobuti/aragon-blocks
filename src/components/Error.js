import styled from "styled-components";

const Error = styled.div`
  border-radius: 4px;
  padding: 24px;
  background-color: #f0a;
  color: white;
`;

export default Error;
