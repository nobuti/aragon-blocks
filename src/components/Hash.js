import React from "react";
import styled from "styled-components";

const Badge = styled.div`
  display: inline-flex;
  border-radius: 4px;
  background-color: #eee;
  padding: 2px 4px;
  font-family: monospace;
`;

const Transaction = ({ hash }) => {
  const shortcut = hash => {
    return hash.length > 13
      ? [hash.substring(0, 6), `...`, hash.substr(-4)].join("")
      : hash;
  };

  const short = hash ? shortcut(hash) : `Invalid transaction`;
  return <Badge title={hash || short}>{short}</Badge>;
};

export default Transaction;
