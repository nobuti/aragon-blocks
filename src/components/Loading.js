import React from "react";
import PropTypes from "prop-types";
import styled, { keyframes } from "styled-components";
import "styled-components/macro";

const bounce = keyframes`
  0%, 100% { 
    transform: scale(0.0);
  } 
  50% { 
    transform: scale(1.0);
  }
`;

const LoadingIndicator = styled.div`
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  position: relative;

  &:after,
  &:before {
    content: "";
    display: block;
    width: 100%;
    height: 100%;
    border-radius: 50%;
    background-color: ${props => props.color};
    opacity: 0.6;
    position: absolute;
    top: 0;
    left: 0;
    animation: ${bounce} 2s infinite ease-in-out;
  }

  &:after {
    animation-delay: -1s;
  }
`;

LoadingIndicator.propTypes = {
  color: PropTypes.string,
  size: PropTypes.number
};

LoadingIndicator.defaultProps = {
  color: "#4a00e0",
  size: 48
};

const Loading = () => (
  <div
    css={`
      padding: 32px;
      display: flex;
      justify-content: center;
    `}
  >
    <LoadingIndicator />
  </div>
);

export default Loading;
