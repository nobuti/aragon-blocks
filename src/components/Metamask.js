import React, { createContext, useState, useEffect, useRef } from "react";
import Web3 from "web3";
import PropTypes from "prop-types";

const ERRORS = {
  denied: "You need to grant account access in order to use this application.",
  forbidden: "You have to install Metamask to use this application."
};

export const MetamaskContext = createContext(null);

export const Metamask = ({ children }) => {
  const [state, setState] = useState({
    web3: null,
    error: null,
    loading: true
  });

  const web3 = useRef();

  useEffect(() => {
    const setupProvider = () => {
      const provider = new Web3.providers.HttpProvider(
        "https://ropsten.infura.io/v3/fbbffc8457ac4c269513bd352c6c5113"
      );
      web3.current = new Web3(provider);
      setState({ web3: web3.current, error: null, loading: false });
    };

    async function trySetWeb3() {
      if (window.ethereum) {
        window.ethereum.autoRefreshOnNetworkChange = false;

        try {
          await window.ethereum.enable();
          setupProvider();
        } catch (e) {
          setState({ web3: null, error: ERRORS.denied, loading: false });
        }
      } else if (window.web3) {
        setupProvider();
      } else {
        setState({ web3: null, error: ERRORS.forbidden, loading: false });
      }
    }

    trySetWeb3();
  }, []);

  return (
    <MetamaskContext.Provider value={state}>
      {children}
    </MetamaskContext.Provider>
  );
};

Metamask.propTypes = {
  children: PropTypes.node
};
