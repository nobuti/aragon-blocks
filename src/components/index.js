export { default as Error } from "./Error";
export { default as Hash } from "./Hash";
export { default as Layout } from "./Layout";
export { default as Loading } from "./Loading";
export { MetamaskContext, Metamask } from "./Metamask";
export { default as RestrictedRoute } from "./RestrictedRoute";
export { default as Table } from "./Table";
