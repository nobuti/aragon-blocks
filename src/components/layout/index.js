export { default as Header } from "./Header";
export { default as Container } from "./Container";
export { default as Hero } from "./Hero";
export { default as Hologram } from "./Hologram";
export { default as Footer } from "./Footer";
export { default as Dataview } from "./Dataview";
