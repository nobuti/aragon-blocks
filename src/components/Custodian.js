import { useContext } from "react";
import PropTypes from "prop-types";

import { MetamaskContext } from "./Metamask";

const Custodian = ({ children }) => {
  const { error, loading } = useContext(MetamaskContext);
  return children({ loading, error });
};

Custodian.propTypes = {
  children: PropTypes.node.isRequired
};

export default Custodian;
