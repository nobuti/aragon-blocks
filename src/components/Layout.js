import React, { useRef } from "react";
import PropTypes from "prop-types";

import { Header, Hero, Container, Hologram, Footer, Dataview } from "./layout";
import Traffic from "./traffic";

const Layout = ({ children }) => {
  const node = useRef();
  const smoothScroll = e => {
    e.preventDefault();
    node.current.scrollIntoView({ behavior: "smooth", block: "start" });
  };

  return (
    <>
      <Hero.Main>
        <Container>
          <Header>CrytoBlocks</Header>
        </Container>

        <Container>
          <Hero.Content>
            <Hero.Left>
              <h2>
                We believe in and
                <br />
                fight for freedom
              </h2>
              <p>
                Monero managed some fundamental analysis after few dump. Someone
                accompanied by a raiden network, therefore, Litecoin stacks lots
                of minimum trustless for the wash trade. Silk Road could be many
                safe crypto-jacking because Augur stacks lots of quick node
                until the cryptocurrency, however, Binance Coin could be some
                address! Since!
              </p>

              <Hero.Action>
                <a onClick={smoothScroll} href="#table">
                  Start exploring transactions
                </a>
              </Hero.Action>
            </Hero.Left>

            <Hero.Right>
              <Hologram />
            </Hero.Right>
          </Hero.Content>
        </Container>
      </Hero.Main>
      <Container>
        <Dataview.Main>
          <Traffic />

          <Dataview.Claim id="dataview" ref={node}>
            <h3>Ten blocks. No more no less</h3>
            <p>Have fun exploring transactions. Since!</p>
          </Dataview.Claim>

          {children}
        </Dataview.Main>
      </Container>

      <Footer>
        Made with{" "}
        <span role="img" aria-label="love">
          ❤️
        </span>{" "}
        by nobuti.
      </Footer>
    </>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired
};

export default Layout;
