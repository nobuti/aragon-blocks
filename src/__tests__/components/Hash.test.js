import React from "react";
import { render } from "@testing-library/react";

import { Hash } from "../../components";

describe("Hash component", () => {
  it("should render properly if hash is provided", () => {
    const hash = "0x686ACf148C1dAaA226b661fff8b7CBbD7c7AEE20";
    const { container, getByTitle } = render(<Hash hash={hash} />);

    expect(container).toMatchSnapshot();
    expect(getByTitle(hash)).toBeDefined();
  });

  it("should render invalid if hash is not provided", () => {
    const { container, getByTitle } = render(<Hash />);

    expect(container).toMatchSnapshot();
    expect(getByTitle("Invalid transaction")).toBeDefined();
  });

  it("should not shortcut the hash if it is not long enough", () => {
    const hash = "0x123456";
    const { container, getByTitle } = render(<Hash hash={hash} />);

    expect(container).toMatchSnapshot();
    expect(getByTitle(hash)).toBeDefined();
    expect(container.innerHTML).toMatch(hash);
  });
});
