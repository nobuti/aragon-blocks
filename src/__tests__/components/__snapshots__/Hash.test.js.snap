// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hash component should not shortcut the hash if it is not long enough 1`] = `
<div>
  <div
    class="sc-bwzfXH ffRHco"
    title="0x123456"
  >
    0x123456
  </div>
</div>
`;

exports[`Hash component should render invalid if hash is not provided 1`] = `
<div>
  <div
    class="sc-bwzfXH ffRHco"
    title="Invalid transaction"
  >
    Invalid transaction
  </div>
</div>
`;

exports[`Hash component should render properly if hash is provided 1`] = `
<div>
  <div
    class="sc-bwzfXH ffRHco"
    title="0x686ACf148C1dAaA226b661fff8b7CBbD7c7AEE20"
  >
    0x686A...EE20
  </div>
</div>
`;
